canvas_modules/harness/cypress/support/canvas/verification-cmds.js

```diff
@@ -1369,9 +1369,21 @@ Cypress.Commands.add("verifyNotificationIconType", (type) => {
 });
 
 Cypress.Commands.add("verifyCanvasTransform", (movString) => {
-	cy.get("#canvas-div-0 .d3-canvas-group")
-		.invoke("attr", "transform")
-		.should("eq", movString);
+	// Verify the argument passed is a string
+	if (movString === String) {
+		cy.get("#canvas-div-0 .d3-canvas-group", { timeout: 10000 })
+			.should("exist")
+			.invoke("attr", "transform")
+			.then((actualTransformString) => {
+				const expectedValues = parseTransformString(movString);
+				const actualValues = parseTransformString(actualTransformString);
+
+				// Compare each part of the transform with compareRange
+				compareCloseTo(actualValues.translateX, expectedValues.translateX);
+				compareCloseTo(actualValues.translateY, expectedValues.translateY);
+				compareCloseTo(actualValues.scale, expectedValues.scale);
+			});
+	}
```

Comment:
I know it ought to be unlikely but, what happens here if `movString` is not a string? Will the function return as if the comparison succeeded? If so that not what we want as we want the test to fail in all error situations.

If you are testing for a string higher up in the function because sometimes zoom.cy.js passes in an undefined value then your code should specifically test for "undefined" and return success and in any other case (other than "string" ) return failure. In fact, since undefined is a value that is specifically passed in by zoom.cy.js, what the function ought to do is make sure the `actualTransformString` is undefined in that situation.

Reply:
I am able to fix this with if,else condition to check whether the movString is undefined, also if movString is not a string the test will fail with below error,
<img width="779" alt="image" src="https://github.com/user-attachments/assets/17fbf17c-5122-409a-9b5b-d2f0e1dad664">